src/text/line.rs

```diff
@@ -646,6 +646,28 @@ fn spans_after_width<'a>(
         })
 }
 
+/// A trait for converting a value to a [`Line`].
+///
+/// This trait is automatically implemented for any type that implements the [`Display`] trait. As
+/// such, `ToLine` shouln't be implemented directly: [`Display`] should be implemented instead, and
+/// you get the `ToLine` implementation for free.
+///
+/// [`Display`]: std::fmt::Display
+pub trait ToLine {
+    fn to_line(&self) -> Line<'_>;
+}
+
+/// # Panics
+///
+/// In this implementation, the `to_line` method panics if the `Display` implementation returns an
+/// error. This indicates an incorrect `Display` implementation since `fmt::Write for String` never
+/// returns an error itself.
+impl<T: fmt::Display> ToLine for T {
+    fn to_line(&self) -> Line<'_> {
+        Line::from(self.to_string())
```

Comment:
This is more a question about idioms than anything else, but why not implement `ToLine` for all `string::ToString` rather than all `fmt::Display`? All this does is call to_string() anyway.

Trying to answer my own question:

1. The docs for `ToString` say don't implement `ToString`, implement `Display`. Counterpoint: that doesn't mean API's shouldn't accept `ToString` when all they do is call `to_string()` on the thing.
2. std::fmt can work with no_std, but std::string can't. Counterpoint: not an issue for Ratatui?
3. There are subtle cultural differences in expectation between `Display` and `ToString`. Counterpoint: I haven't found any.  ;-)

Reply:
1st and 3rd answers makes the most sense to me and that's probably what I'd come up with too here.

Reply:
I suspect that it's rare to see `<T: ToString>`, though I'm not sure which is correct. The thing to note is that if the ToString case is correct on seeing this in the real world, then changing the blanket implementation to be against ToString is a 100% compatible change for all callers, the other direction is not however.